cql3/expr/expression.cc

```diff
@@ -370,6 +374,15 @@ bool_or_null limits(const expression& lhs, oper_t op, const expression& rhs, con
             return false; // both NULL
         }
         break;
+    case oper_t::IN_SPECIAL_NULL:
+        if (!sides_bytes.second.has_value()) {
+            // Nothing can be IN a list if there is no list.
+            return false;
+        }
```

Comment:
I changed `is_one_of()` (which now contains `is_one_of_special_null()`) to return `FALSE` for this case, to be more consistent with LWT, which doesn't return NULL (though it's rather invisible, since the condition is consumed internally and not exposed).